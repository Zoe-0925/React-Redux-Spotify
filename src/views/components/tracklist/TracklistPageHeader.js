import React, { Fragment } from 'react';
import { ReactComponent as PlayIcon } from "../../../svgs/play.svg"
import { ReactComponent as HeartIcon } from '../../../svgs/heart.svg'
import PropTypes from 'prop-types';
import { Icon } from 'semantic-ui-react'
import 'semantic-ui-css/semantic.min.css';

export default function TracklistPageHeader(props) {

    return (
        <Fragment>
            <div className="playlistPageInfo">
                <div className="playlistPageImage">
                    {props.imgSrc && <img src={props.imgSrc} alt="track-card" key="card-img" />}
                </div>

                <div className="playlistPageContent">
                    <p className="smallText uppercase bold" style={{ color: "white" }}>{props.title}</p>
                    <h1 style={{ color: "white", fontSize: "40px" }}>{props.albumName}</h1>

                    <div className="playlistPageDesc">
                        <p className="spotify" style={{ color: "white" }}>{props.artistName}</p>
                    </div>
                </div>
            </div>
            <div className="playlistPageSongs">
                <div className="playlistButtons">
                    <span className="playIcon" onClick={props.playAlbum}>
                        <PlayIcon />
                    </span>
                    <div className="icons">
                        {props.saved &&
                            <div className="icon iconsHeart" onClick={props.toggleSave}>
                                <Icon name='heart' className="coloredIcon" size='big' />
                            </div>}
                        {!props.saved &&
                            <div className="icon iconsHeart" onClick={props.toggleSave}>
                                <Icon name='heart outline' className="coloredIcon" size='big' />
                            </div>
                        }
                        <div className="icon iconsDots"></div>
                    </div>
                </div>
            </div>
        </Fragment>
    )
}

TracklistPageHeader.prototype = {
    saveToLikes: PropTypes.func,
    playAlbum: PropTypes.func,
    artistName: PropTypes.string,
    albumName: PropTypes.string,
    imgSr: PropTypes.string,
    title: PropTypes.string,
}