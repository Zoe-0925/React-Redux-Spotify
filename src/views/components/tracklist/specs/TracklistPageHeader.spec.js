import React from 'react';
import renderer from 'react-test-renderer';
import TracklistPageHeader from "../TracklistPageHeader"
import { shallow } from 'enzyme';

const imgSrc = "https://i.scdn.co/image/ab67616d0000b2736040effba89b9b00a6f6743a"
const artistName = "Lady Gaga"
const title = "mock title"
const albumName = "album Name"
const playAlbum = jest.fn();
const saveToLikes = jest.fn();

//TODO find play icon and simulate click,
//expect  playAlbum() to be called

//TODO find icon iconsHeart and simulate click,
//expect  saveToLikes() to be called

describe('TracklistPageHeader Component', () => {
    it('renders correctly', () => {
        const component = renderer.create(<TracklistPageHeader imgSrc={imgSrc} albumName={albumName}
            artistName={artistName} title={title} playAlbum={playAlbum} saveToLikes={saveToLikes} />);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should dispatch an action on button click', () => {
        const wrapper = shallow((<TracklistPageHeader imgSrc={imgSrc} albumName={albumName}
            artistName={artistName} title={title} playAlbum={playAlbum} saveToLikes={saveToLikes} />))
        wrapper.find('.iconsHeart').simulate('click');
        wrapper.find('.playIcon').simulate('click');
        expect(playAlbum).toBeCalled();
        expect(saveToLikes).toBeCalled();
    })
}
)