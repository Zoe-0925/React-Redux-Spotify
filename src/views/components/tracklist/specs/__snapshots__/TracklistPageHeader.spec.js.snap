// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TracklistPageHeader Component renders correctly 1`] = `
Array [
  <div
    className="playlistPageInfo"
  >
    <div
      className="playlistPageImage"
    >
      <img
        alt="track-card"
        src="https://i.scdn.co/image/ab67616d0000b2736040effba89b9b00a6f6743a"
      />
    </div>
    <div
      className="playlistPageContent"
    >
      <p
        className="smallText uppercase bold"
        style={
          Object {
            "color": "white",
          }
        }
      >
        mock title
      </p>
      <h1
        style={
          Object {
            "color": "white",
            "fontSize": "40px",
          }
        }
      >
        album Name
      </h1>
      <div
        className="playlistPageDesc"
      >
        <p
          className="spotify"
          style={
            Object {
              "color": "white",
            }
          }
        >
          Lady Gaga
        </p>
      </div>
    </div>
  </div>,
  <div
    className="playlistPageSongs"
  >
    <div
      className="playlistButtons"
    >
      <span
        className="playIcon"
        onClick={[MockFunction]}
      >
        <svg>
          play.svg
        </svg>
      </span>
      <div
        className="icons"
      >
        <div
          className="icon iconsHeart"
          onClick={[MockFunction]}
        >
          <svg
            id="heart-icon"
          >
            heart.svg
          </svg>
        </div>
        <div
          className="icon iconsDots"
        />
      </div>
    </div>
  </div>,
]
`;
